tools/update_runnablejobs.py

```diff
@@ -144,44 +151,15 @@ def parse_platform(platform):
     return platform
 
 
-def add_jobs_to_jobpriority(new_data=None, priority=1, timeout=0, set_expired=False):
+def _update_database_with_new_jobs(new_data, set_expiration=False):
+    ''' Return jobs that are new.'''
+    # XXX: jmaher: we might want to change how this method is implemented to do directly query the
+    #              DB by creating a good index. It is probably faster than iterating over rows
+    # XXX: jmaher: I would like to change this method to return a list of jobs to be stored rather
+    #              than do it in the middle of the for loop.
     added_jobs = []
-
-    if not new_data:
-        return
-
-    # This is a loop to remove duplicates (including buildsystem -> * if needed)
-    # By doing this it allows us to have a single database query
-    no_duplicates = []
-    for job in new_data['results']:
-        platform = parse_platform(job['build_platform'])
-        if platform == None or platform == "":
-            continue
-
-        testtype = parse_testtype(job['build_system_type'],
-                                  job['ref_data_name'],
-                                  job['platform_option'],
-                                  job['job_type_name'])
-        if testtype == None or testtype == "":
-            continue
-
-        job["platform"] = platform
-        job["testtype"] = testtype
-        found = False
-        for nd in no_duplicates:
-            if job['testtype'] == nd['testtype'] and \
-               job['platform'] == nd['platform'] and \
-               job['platform_option'] == nd['platform_option']:
-                found = True
-                break
-
-        if found:
-            if job['build_system_type'] != nd['build_system_type']:
-                job['build_system_type'] = '*'
-        no_duplicates.append(job)
-    new_data['results'] = no_duplicates
```

Comment:
the duplicate removal is pretty key to quicker performance when dealing with the database